packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -1591,8 +1626,49 @@ export class CacheDataClient implements IDataClient {
     }
 
     return await this.rateLimited(async () => {
-      const itemsToUse = this.convertSetBatchElements(items);
-      const ttlToUse = ttl || this.defaultTtlSeconds;
+      let itemsToUse: [Uint8Array, Uint8Array][] =
+        this.convertSetBatchElements(items);
+      if (options?.compress) {
+        this.logger.trace(
+          'CacheClient.setBatch; compression enabled, calling value compressor'
+        );
+        if (this.valueCompressor === undefined) {
+          return this.cacheServiceErrorMapper.returnOrThrowError(
+            new CompressionError('CacheClient.setBatch', 'compress'),
+            err => new CacheSetBatch.Error(err)
+          );
+        }
+        const compressor: ICompression = this.valueCompressor;
+        itemsToUse = await Promise.all(
+          itemsToUse.map(async ([key, value]) => {
+            return [
+              key,
+              await compressor.compress(
```

Comment:
Have we thought about how the additional CPU contention from compression will interact with servicing other batch requests?

Reply:
have thought about it but not measured it. At the end of the day i think it's going to be very specific to a given user / use case as to whether this is a good tradeoff.

In the future if we want to set up some mechanism for limiting concurrency here, we can; in general I'm pretty opposed to code like this that spawns off an unknown number of async tasks. But I didn't want to add complexity to the config/API until we have some data from the wild.